.github/workflows/docs/build-and-push.md

```diff
@@ -91,13 +91,42 @@ jobs:
   build:
     secrets: inherit
     permissions:
-     contents: read
-     id-token: write 
-     packages: write
-   uses: mozilla-it/deploy-actions/.github/workflows/build-and-push.yml@main
-   with:
-     image_name: your-service
-     gar_name: your-tenant-prod
-     project_id: moz-fx-you-tenant-prod
-     should_tag_ghcr: true
-```
\ No newline at end of file
+      contents: read
+      id-token: write
+      packages: write
+    uses: mozilla-it/deploy-actions/.github/workflows/build-and-push.yml@main
+    with:
+      image_name: your-service
+      gar_name: your-tenant-prod
+      project_id: moz-fx-you-tenant-prod
+      should_tag_ghcr: true
+```
+
+## Troubleshooting
+
+Is your build failing? Here are some steps to follow to ensure your setup is
+correct.
+
+In your tenant bootstrap, ensure the following are defined:
+
+1. `github_build_repositories = ["<your github org>/<your github repo>"]`,
+   and
+1. `github_require_build_env = false`.
+
+Examples: [enabling `github_build_repositories`][github_build_repositories], [disabling `github_require_build_env`][github_require_build_env].
+
+[github_build_repositories]: https://github.com/mozilla/webservices-infra/pull/6778/files#diff-ceb2f406e3b4bdf08b28c2287a8f099f3416b9169fe730a53eeb721ec8fc5139R9-R11
+[github_require_build_env]: https://github.com/mozilla/webservices-infra/pull/6779/files#diff-ceb2f406e3b4bdf08b28c2287a8f099f3416b9169fe730a53eeb721ec8fc5139R12
+
+In your GitHub organization or repository, ensure the following environment
+variables are defined:
+
+1. `GCPV2_WORKLOAD_IDENTITY_POOL_PROJECT_NUMBER`, and
+1. `GCPV2_GITHUB_WORKLOAD_IDENTITY_PROVIDER`.
```

Comment:
It's good to let people know about these gotchas, but we also have this documented [here](https://mozilla-hub.atlassian.net/wiki/spaces/SRE/pages/997163545/How+to+Publish+Container+Images+to+GAR). Can we keep the troubleshooting section, but link to confluence for these specific gotchas? Just so we only need to update things in one, central place.

Reply:
Done!